src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1354,6 +1356,8 @@ ObjectMonitor* ObjectSynchronizer::inflate(Thread* current, oop object,
     m->set_header(mark);
 
     if (object->cas_set_mark(markWord::encode(m), mark) != mark) {
+      // Release object's oop storage since we don't need this ObjectMonitor:
+      m->release_object();
```

Comment:
Just to be clear, inflate() is called when a JavaThread is
in VM (or by the VM Thread) so it is safe to call release.